### PR TITLE
Ingest multiple vectorsets

### PR DESCRIPTION
**Pull request by jotare**
### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nucliadb/nucliadb/common/ids.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+class ParagraphId:
+    rid: str
+    field_id: str
+    paragraph_start: int
+    paragraph_end: int
+    subfield_id: Optional[str] = None
+
+    def full(self) -> str:
+        if self.subfield_id is None:
+            return f"{self.rid}/{self.field_id}/{self.paragraph_start}-{self.paragraph_end}"
```
Comment (javitonino) [inline nucliadb/nucliadb/common/ids.py:56]: Maybe this should store a `FieldId` instead of `rid`, `field_id` and `subfield_id` separately. Then you can do `return f"{self.field.full()}/{self.paragraph_start}-{self.paragraph_end}`.

I think it's more composable this way, represents the hierarchy and we keep the `if subfield_id` contained to the `FieldId` instead of spreading it everywhere.

File: nucliadb/nucliadb/ingest/tests/unit/service/test_writer.py [hunk 2]
```diff
@@ -668,6 +668,10 @@ async def test_Index(self, writer: WriterServicer):
                 "nucliadb.ingest.service.writer.get_transaction_utility",
                 MagicMock(return_value=txn),
             ),
+            patch(
```
Comment (javitonino) [inline nucliadb/nucliadb/ingest/tests/unit/service/test_writer.py:671]: Why is this needed now? Or, alternatively, why wasn't it needed before?

File: nucliadb/nucliadb/common/ids.py (hunk 1 above)
Comment (jotare) [reply nucliadb/nucliadb/common/ids.py:56]: Sure, it'll be cleaner!

File: nucliadb/nucliadb/ingest/tests/unit/service/test_writer.py (hunk 2 above)
Comment (jotare) [reply nucliadb/nucliadb/ingest/tests/unit/service/test_writer.py:671]: This is a too-mocked-test. Before my changes, the resource was a mock and the methods to build the broker message weren't even called. Now we are using the resource to build the broker message, so there's a bunch of things that are actually accessed. As this test is almost useless and testing the ingest gRPC, I decided to mock the broker message generation

File: nucliadb/nucliadb/ingest/orm/brain.py [hunk 3]
```diff
@@ -268,17 +268,18 @@ def apply_field_vectors(
         for subfield, vectors in vo.split_vectors.items():
             # For each split of this field
             for index, vector in enumerate(vectors.vectors):
-                paragraph_key = ids.ParagraphId(
+                _field_id = ids.FieldId(
```
Comment (javitonino) [inline nucliadb/nucliadb/ingest/orm/brain.py:271]: You can create this outside of this loop.

File: nucliadb/nucliadb/ingest/orm/brain.py [hunk 4]
```diff
@@ -293,15 +294,17 @@ def apply_field_vectors(
                 )
 
         for index, vector in enumerate(vo.vectors.vectors):
-            paragraph_key = ids.ParagraphId(
+            _field_id = ids.FieldId(
```
Comment (javitonino) [inline nucliadb/nucliadb/ingest/orm/brain.py:297]: Same here.

Comment (jotare) [reply nucliadb/nucliadb/ingest/orm/brain.py:297]: Thanks!

File: nucliadb/nucliadb/common/datamanagers/fields.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+PbFieldTypes = Union[
+    resources_pb2.Conversation,
+    resources_pb2.FieldDatetime,
+    resources_pb2.FieldFile,
+    str,
+    resources_pb2.FieldKeywordset,
+    resources_pb2.FieldLayout,
+    resources_pb2.FieldLink,
+    resources_pb2.FieldText,
+]
```
Comment (lferran) [inline nucliadb/nucliadb/common/datamanagers/fields.py:38]: Is this used anywhere?

File: nucliadb/nucliadb/common/datamanagers/resources.py [hunk 6]
```diff
@@ -26,26 +26,33 @@
...
-from nucliadb.ingest.orm.resource import Resource as ResourceORM
 from nucliadb.ingest.settings import settings as ingest_settings
-from nucliadb_protos import noderesources_pb2, resources_pb2, writer_pb2
+from nucliadb_protos import noderesources_pb2, resources_pb2
 from nucliadb_utils.utilities import get_storage
 
 from .utils import with_transaction
 
+if TYPE_CHECKING:
+    from nucliadb.ingest.orm.resource import Resource as ResourceORM
```
Comment (lferran) [inline nucliadb/nucliadb/common/datamanagers/resources.py:36]: not used I think

**Review by lferran — APPROVED** [review]

File: nucliadb/nucliadb/common/datamanagers/fields.py (hunk 5 above)
Comment (jotare) [reply nucliadb/nucliadb/common/datamanagers/fields.py:38]: Oh, not anymore. I'll remove it

File: nucliadb/nucliadb/common/datamanagers/resources.py (hunk 6 above)
Comment (jotare) [reply nucliadb/nucliadb/common/datamanagers/resources.py:36]: It is, in line 347

File: nucliadb/nucliadb/tests/integration/test_vectorsets.py [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        query_pb = nodereader_pb2.SearchRequest(
+            shard=shard_id,
+            body="this is a query for my vectorset",
+            vector=[1.23] * dimension,
+            vectorset=vectorset,
+            result_per_page=5,
+        )
+        with pytest.raises(Exception) as exc:
+            results = await node.reader.Search(query_pb)  # type: ignore
+            assert "inconsistent dimensions" in str(exc).lower()
```
Comment (lferran) [inline nucliadb/nucliadb/tests/integration/test_vectorsets.py:83]: I think you're swallowing any exceptions here and the assert is not happening. You need to de-indent the assert one level

Comment (jotare) [reply nucliadb/nucliadb/tests/integration/test_vectorsets.py:83]: Oh true! Thanks!
